src/smart-components/Message/context/MessageProvider.tsx

```diff
@@ -0,0 +1,45 @@
+// todo@v4.0.0: combine with the provider in core-ts, see:
+// https://github.com/sendbird/sendbird-uikit-core-ts
+// packages/react-uikit-message-template-view/src/context/MessageContextProvider.tsx
+import React from 'react';
+import { BaseMessage } from '@sendbird/chat/message';
+
+export type MessageProviderProps = {
+  children: React.ReactNode;
+  message: BaseMessage;
```

Comment:
It will be `BaseMessage | UserMessage | FileMessage | AdminMessage`, no?

Reply:
We can discuss the typing, I have couple of ideas